test/framework/src/main/java/org/opensearch/test/AbstractQueryTestCase.java

```
@@ -868,4 +869,26 @@ public void testCacheability() throws IOException {
         assertTrue("query should be cacheable: " + queryBuilder.toString(), context.isCacheable());
     }
 
+    /**
+     * Check if a filter can be applied to the abstract query builder.
+     * @throws IOException
+     */
+    public void testFilter() throws IOException {
+        QB queryBuilder = createTestQueryBuilder();
+        // Test for Null Filter case
+        QueryBuilder returnedQuerybuilder = queryBuilder.filter(null, null);
+        assertEquals(queryBuilder, returnedQuerybuilder);
+
+        // Test for OR or IGNORE_IF_EXISTS case
+        returnedQuerybuilder = queryBuilder.filter(QueryBuilders.matchAllQuery(), QueryBuilder.FilterCombinationMode.OR);
```

We don't plan to support OR or IGNORE_IF_EXISTS case now so don't define them in the enum or you should throw unsupported operation exception.